app/org/sagebionetworks/bridge/dynamodb/DynamoFPHSExternalIdentifierDao.java

```diff
@@ -0,0 +1,112 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.Resource;
+
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.dao.FPHSExternalIdentifierDao;
+import org.sagebionetworks.bridge.exceptions.EntityAlreadyExistsException;
+import org.sagebionetworks.bridge.exceptions.EntityNotFoundException;
+import org.sagebionetworks.bridge.models.accounts.ExternalIdentifier;
+import org.sagebionetworks.bridge.models.accounts.FPHSExternalIdentifier;
+import org.springframework.stereotype.Component;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper.FailedBatch;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBScanExpression;
+
+@Component
+public class DynamoFPHSExternalIdentifierDao implements FPHSExternalIdentifierDao {
+
+    private DynamoDBMapper mapper;
+    
+    @Resource(name = "fphsExternalIdDdbMapper")
+    public final void setMapper(DynamoDBMapper mapper) {
+        this.mapper = mapper;
+    }
+    
+    @Override
+    public boolean verifyExternalId(ExternalIdentifier externalId) {
+        checkNotNull(externalId);
+        
+        DynamoFPHSExternalIdentifier record = getExternalId(externalId, false);
+        return (record != null && !record.getRegistered());
+    }
+
+    @Override
+    public void registerExternalId(ExternalIdentifier externalId) {
+        checkNotNull(externalId);
+        
+        DynamoFPHSExternalIdentifier record = getExternalId(externalId, true);
+        record.setRegistered(true);
+        mapper.save(record);
+    }
+    
+    @Override
+    public void unregisterExternalId(ExternalIdentifier externalId) {
+        checkNotNull(externalId);
+        
+        DynamoFPHSExternalIdentifier record = getExternalId(externalId, true);
+        record.setRegistered(false);
+        mapper.save(record);
+    }
```

Comment:
We should add unit tests for unregister() with all the same cases as register(). Reading through the code, it looks like this wouldn't work, because getExternalId() throws if the ID is already registered.